front/app/containers/CustomPageShow/index.tsx

```diff
@@ -124,6 +139,13 @@ const CustomPageShow = () => {
               <FileAttachments files={remotePageFiles} />
             </AttachmentsContainer>
           )}
+        {pageAttributes.events_widget_enabled &&
+          projectIds &&
+          projectIds.length > 0 && (
+            <ContentContainer>
+              <EventsWidget projectIds={projectIds} />
+            </ContentContainer>
+          )}
```

Comment:
These checks prevent the EventsWidget from loading all projects when there are no projectIds yet (and shortly flashing them on the screen). Not ideal having to deal with this on an individual basis. The first time we render EventsWidget, we already should have our projectIds. This can be achieved by attaching projectIds to the page data and check for the page to be isNilOrError.